examples/lighting-app/nrfconnect/main/AppTask.cpp

```diff
@@ -426,13 +389,25 @@ void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
     }
 }
 
-#ifdef CONFIG_CHIP_NFC_COMMISSIONING
 void AppTask::ChipEventHandler(const ChipDeviceEvent * event, intptr_t /* arg */)
 {
     if (event->Type != DeviceEventType::kCHIPoBLEAdvertisingChange)
         return;
 
-    if (event->CHIPoBLEAdvertisingChange.Result == kActivity_Started)
+    if (event->CHIPoBLEAdvertisingChange.Result == kActivity_Stopped)
+    {
+#ifdef CONFIG_CHIP_NFC_COMMISSIONING
+        NFCMgr().StopTagEmulation();
+#endif
+#ifdef CONFIG_MCUMGR_SMP_BT
+        // After CHIPoBLE advertising stop, start advertising SMP in case Thread is enabled or there are no active CHIPoBLE
+        // connections (exclude the case when CHIPoBLE advertising is stopped on the connection time)
+        if (GetDFUOverSMP().IsEnabled() && (ConnectivityMgr().IsThreadProvisioned() || ConnectivityMgr().NumBLEConnections() == 0))
```

Comment:
If I understand this and the PR description, I believe `IsThreadProvisioned()` is being used as a proxy to determine  whether the device is commissioned to a Matter fabric.

If so, that seems like a fragile assumption.  This would definitely break if the device is manually joined to a Thread network through the debug console before proceeding to commissioning (which otherwise is a useful thing to do).  

Is there a more formal way for us to determine whether this node is commissioned to one or more fabrics?  That is ultimately the thing we want to branch on, correct?

Reply:
Just to clarify, do you mean the situation that we join the Thread network manually and because of that IsThreadProvisioned() condition is met, even though we are actually not commissioned, so the pairing window will not be open? If yes, I think that's a valid point. Moreover we already use the same check in other places like here: https://github.com/project-chip/connectedhomeip/blob/master/src/app/server/Server.cpp#L522

Do you think that better option would be checking IsFullyProvisioned() condition? Currently with that one TODO: https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/internal/GenericConfigurationManagerImpl.cpp#L923 it doesn't seem to make much difference ...

Reply:
Going even further I think that currently most platforms doesn't allow to trigger starting BLE advertising basing on the IsThreadProvisioned() condition: https://github.com/project-chip/connectedhomeip/blob/master/examples/lock-app/qpg/src/AppTask.cpp#L342

Reply:
> Just to clarify, do you mean the situation that we join the Thread network manually and because of that IsThreadProvisioned() condition is met, even though we are actually not commissioned, so the pairing window will not be open? If yes, I think that's a valid point.

Yes, exactly.

> Do you think that better option would be checking IsFullyProvisioned() condition? Currently with that one TODO: https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/internal/GenericConfigurationManagerImpl.cpp#L923 it doesn't seem to make much difference ...

I suppose yes.  But `IsFullyProvisioned()` is provided by the configuration manager and may need some work.  It definitely seems odd in the context of multi-admin.  Maybe that's just naming.  Maybe it should be called `IsJoinedToOneOrMoreFabrics()`.  I'm sure code would would need to be tweaked though to make this evaluate correctly in multi-admin scenarios.

> I created an issue to track this one, as I believe that fixing all occurences in the code is a bit out of this PR scope: #8897

Sounds good.  Thank you!